### DISCUSSION TITLE
make error: fatal error: libmesh/mesh_netgen_interface.h: No such file or directory

### DISCUSSION THREAD

**Discussion by frankwugeo** [idaholab/moose#28987] [Q&A Getting Started]
### Check these boxes if you have followed the posting rules.

- [X] I have consulted the [Posting Guidelines](https://github.com/idaholab/moose/discussions/18270).
- [X] I have searched the [Discussions Forum](https://github.com/idaholab/moose/discussions) and [MOOSE Framework Troubleshooting](https://mooseframework.inl.gov/help/troubleshooting.html) and have not found what I was looking for
- [X] Q&A Getting Started is the most appropriate category for my question (trouble installing, beginner user, ...)

### Issue or question about MOOSE

Hi all,

I am installing MOOSE and RACCOON. After building MOOSE, then clone RACCOON, and run below to get the submodule.

```
git submodule update --init --recursive
``` 
Then I try to compile RACCOON using make, but got the following error.

I also checked and found 'mesh_netgen_interface.h' file in the right directory. But for the 'XYZDelaunayGenerator' here, I did not find it in a successfully built MOOSE file on another pc of mine. 
```
in /projects/raccoon/moose/framework/src/meshgenerators
``` 

Many thanks for any help.

Best

### (Optional) code in question / simulation log / errors

```shell
In file included from /home/user/projects/raccoon/moose/framework/build/unity_src/meshgenerators_Unity.C:85:
/home/user/projects/raccoon/moose/framework/src/meshgenerators/XYZDelaunayGenerator.C:19:10: fatal error: libmesh/mesh_netgen_interface.h: No such file or directory
   19 | #include "libmesh/mesh_netgen_interface.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make: *** [/home/user/projects/raccoon/moose/framework/build.mk:150: /home/user/projects/raccoon/moose/framework/build/unity_src/meshgenerators_Unity.x86_64-conda-linux-gnu.opt.lo] Error 1
```

**Comment by GiudGiud** [comment]
Hello

> But for the 'XYZDelaunayGenerator' here, I did not find it in a successfully built MOOSE file on another pc of mine.

It was added only a few month ago so maybe your other version of MOOSE did not have it.

Most likely, your version of libmesh is not up to date. 

How are you installing moose? Are you using conda to get moose-petsc/libmesh? Or are you running the scripts to get petsc/libmesh?

**Reply by GiudGiud** [reply]
Quick and dirty potential solution could be (if you are in a hurry)
```
cd ~/projects/racoon/moose/framework
rm src/meshgenerators/XYZDelaunayGenerator.C
rm include/meshgenerators/XYZDelaunayGenerator.h
```
then build

**Reply by frankwugeo** [reply]
Thanks for the reply.

> Most likely, your version of libmesh is not up to date.

I run the scripts again to get petsc/libmesh.

And get the error when update libmesh:
```
/usr/include/features-time64.h:21:10: fatal error: bits/timesize.h: No such file or directory
   21 | #include <bits/timesize.h>
      |          ^~~~~~~~~~~~~~~~~
compilation terminated.
``` 

For the way that remove XYZDelaunayGenerator and build, another error related to libmesh pops up:
```
In file included from /home/user/projects/raccoon/moose/framework/build/unity_src/postprocessors_Unity.C:99:
/home/user/projects/raccoon/moose/framework/src/postprocessors/Residual.C: In member function 'virtual libMesh::Real Residual::getValue() const':
/home/user/projects/raccoon/moose/framework/src/postprocessors/Residual.C:49:5: error: 'LibmeshPetscCall' was not declared in this scope
   49 |     LibmeshPetscCall(SNESGetFunctionNorm(snes, &norm));
      |     ^~~~~~~~~~~~~~~~
```

**Reply by GiudGiud** [reply]
The second error means you can't move forward without updating libmesh. 

The first error is a missing system library. This is odd. You could just add it manually, it s a very simple one. 

You are compiling on Linux right?
